### PR TITLE
[CAPI] Expose Layer Enums

### PR DESCRIPTION
**Pull request by jijoongmoon**
This patch contains layer enums to be exposed.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

**Review by songgot — APPROVED** [review]
Look good to me

File: api/nntrainer-api-common.h [hunk 1]
```diff
@@ -59,7 +59,14 @@ typedef enum {
                                        Sigmoid Loss Layer type (Since 6.5) */
   ML_TRAIN_LAYER_TYPE_LOSS_CROSS_ENTROPY_SOFTMAX = 502, /**< Cross Entropy with
                                        Softmax Loss Layer type (Since 6.5) */
-  ML_TRAIN_LAYER_TYPE_UNKNOWN = 999                     /**< Unknown Layer */
+  ML_TRAIN_LAYER_TYPE_LSTMCELL = 601, /**< LSTM Cell Layer type (Since 7.0) */
+  ML_TRAIN_LAYER_TYPE_GRUCELL = 602,  /**< GRU Cell Layer type (Since 7.0) */
+  ML_TRAIN_LAYER_TYPE_RNNCELL = 603,  /**< RNN Cell Layer type (Since 7.0) */
+  ML_TRAIN_LAYER_TYPE_ZONEOUTLSTMCELL =
+    604, /**< ZoneoutLSTM Cell Layer type (Since 7.0) */
+  ML_TRAIN_LAYER_TYPE_CONV1D = 19, /**< Convolution 1D Layer type (Since 7.0) */
```
Comment (myungjoo) [inline api/nntrainer-api-common.h:67]: Do you want to reorder? (to LINE 50)

**Review by jihochu — APPROVED** [review]
LGTM

File: api/nntrainer-api-common.h (hunk 1 above)
Comment (jijoongmoon) [reply api/nntrainer-api-common.h]: fixed.

**Review by lhs8928 — APPROVED** [review]
LGTM

**Pull request merged**